### ISSUE TITLE
Postcode - set the minimum length for html5 forms

### ISSUE THREAD

**Issue by ShaiMagal** [PrestaShop/PrestaShop#37303]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Our postcode input looks like:
`<input id="field-postcode" class="form-control" name="postcode" type="text" value="12345" maxlength="12" required="">`
So, we use maxlength to tell browser 12 is max length, but we don't use minlength.
We should use this, because if customer wrote short postcode, he will be stopped by browser and we prevent unnecessary reload
If postcode format should be NNNNN, we must set minlength="5".
If postcode format should be NNNN, we must set minlength="4".
etc...
Example:
`<input id="field-postcode" class="form-control" name="postcode" type="text" value="12345" minlength="5" maxlength="12" required="">`

### Expected behavior

Use minlength

### Steps to reproduce

1. login to user account
2. go to section address
3. edit any adress
4. fill there "1234" to postcode (where is format for example NNNNN)
5. click on submit
6. you will see error about postcode, but AFTER reload (sub,ot). If we use minlength, customer will see error earlier and submit will not be processed before the customer fixes it

### PrestaShop version(s) where the bug happened

9.0.0

### PHP version(s) where the bug happened

8.3

### Your company or customer's name goes here (if applicable).

https://www.openservis.cz/

**Comment by florine2623** [comment]
Hello,

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you create it. 

If you or anyone else is willing to [contribute this feature](https://devdocs.prestashop.com/8/contribute/contribute-pull-requests/) to the PrestaShop Project, the teams will be happy to help by ensuring that it fits well within the project. Please let us know if that is the case!

Thank you

**Comment by ibahloul-ps** [comment]
Hello @ShaiMagal 

Thanks for your report.
I don't understand the last bullet point, when I try to reproduce with your scenario, here is the result : 
![Image](https://github.com/user-attachments/assets/f5248678-5d77-4383-8e83-a966ec8c968f)

What should be the expected behavior from your point of view? An information message directly displayed by the browser instead of these error messages? 🤔

**Comment by ShaiMagal** [comment]
@ibahloul-ps yes, because this customer will fix this before he submit form :)
It is for better user experience.

**Comment by ShaiMagal** [comment]
@ibahloul-ps 
Any movement here?
It prevent for reload page, if there is potential error in postcode.

**Comment by ShaiMagal** [comment]
@ibahloul-ps

**Comment by ShaiMagal** [comment]
bump

**Comment by Progi1984** [comment]
Fixed in #38903 .
